node/test/translators/tA.js

```diff
@@ -0,0 +1,71 @@
+"use strict";
+// This code uses ES2015 syntax that requires at least Node.js v4.
+// For Node.js ES2015 support details, reference http://node.green/
+
+const EventEmitter = require("events");
+
+/**
+ * The translator class does not directly implement any device interfaces, but has
+ * one required method for creating a device instance.
+ */
+class TestTranslatorA {
+    createDevice(deviceProps) {
+        return new TestDeviceA(deviceProps);
+    }
+}
+
+// Export an instance of the translator from the module.
+module.exports = new TestTranslatorA();
+
```

Comment:
Instead of exporting a translator object with a createDevice() method, I considered a simpler approach of just exporting the device class (constructor function). However, I think I prefer this model because it allows the translator to instantiate different classes of device depending on the deviceProps, and also because there is an explicit ITranslator interface that is being implemented, rather than just a convention to follow where the constructor takes a device props.